src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs

```diff
@@ -2013,16 +2014,10 @@ public void Append(uint hashcode, Vertex element)
             _Entries.Add(new Entry(hashcode, element));
         }
 
-        // Returns 1 + log2(x) rounded up, 0 iff x == 0
+        // Calculates the highest bit set in a given unsigned integer.
         private static int HighestBit(uint x)
         {
-            int ret = 0;
-            while (x != 0)
-            {
-                x >>= 1;
-                ret++;
-            }
-            return ret;
+            return 1 << (sizeof(uint) * 8) - BitOperations.LeadingZeroCount(x);
```

Comment:
nit: to clarify operator precedence:
```suggestion
            return 1 << ((sizeof(uint) * 8) - BitOperations.LeadingZeroCount(x));
```